iocage_lib/ioc_json.py

```diff
@@ -162,6 +162,85 @@ def __write_file(self):
             )
 
 
+class IOCCpuset(object):
+
+    def __init__(self, name):
+        self.jail_name = f'ioc-{name}'
+
+    def set_cpuset(self, value=None):
+        if not value:
+            value = 'all'
+
+        failed = False
+        try:
+            iocage_lib.ioc_exec.SilentExec(
+                ['cpuset', '-l', value, '-j', self.jail_name],
+                None, unjailed=True, decode=True
+            )
+        except iocage_lib.ioc_exceptions.CommandFailed:
+            failed = True
+        finally:
+            return failed
+
+    @staticmethod
+    def retrieve_cpu_sets():
+        cpu_sets = -2
+        try:
+            output = iocage_lib.ioc_exec.SilentExec(
+                ['cpuset', '-g', '-s', '0'],
+                None, unjailed=True, decode=True
+            )
+        except iocage_lib.ioc_exceptions.CommandFailed:
+            pass
+        else:
+            result = re.findall(
+                r'.*mask:.*(\d+)$',
+                output.stdout.split('\n')[0]
+            )
+            if result:
+                cpu_sets = int(result[0])
+        finally:
+            return cpu_sets
+
+    @staticmethod
+    def validate_cpuset_prop(value, raise_error=True):
+        failed = False
+        cpu_sets = IOCCpuset.retrieve_cpu_sets() + 1
+
+        if not any(
+            cond for cond in (
+                re.findall(
```

Comment:
I fail to understand what's going on here, would you mind adding a few unit tests to this method to demonstrate and verify all the cases that can happen?

Reply:
@themylogin I am validating the prop wrt the 4 rules mentioned at line 232 in my changes. And no worries, I can add a few unit tests to demonstrate the scenarios. In words for the first regex, they are as follows:
1) Make sure user doesn't duplicate cpu numbers
2) The order wrt comma's is always correct i.e `0,2` and not `0,2,`

Reply:
@themylogin I have added a few scenarios for functional tests as those can be tested in CI as well. Please have a look and let me know if any changes are required. Thank you

Reply:
@sonicaj I suggested unit tests specifically targeted the cpuset value parser, because it's complicated write-only regex. I don't mind them if they work, it's just I want to be safe if I'll need to change them in the future.

Reply:
@themylogin no worries, i'll have the PR updated wrt that.

Reply:
@themylogin I've updated the PR, please have a look and let me know if i could perhaps improve the tests ? Thank you